fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -673,20 +703,32 @@ private void maybeLogBuildProgress(SubspaceProvider subspaceProvider, Tuple star
      */
     @Nonnull
     public CompletableFuture<Key.Evaluated> buildUnbuiltRange(@Nonnull FDBRecordStore store, @Nullable Key.Evaluated start, @Nullable Key.Evaluated end) {
-        return buildUnbuiltRange(store, convertOrNull(start), convertOrNull(end))
+        return buildUnbuiltRange(store, start, end, totalRecordsScanned);
```

Comment:
It seems a little weird to me that this uses `totalRecordsScanned`. I suppose the alternative is to make `recordsScanned` nullable and then not increment it if it's `null`, which maybe isn't any better. I guess its contract is documented in the log function, so maybe it's fine.

Reply:
I could also pass in `new AtomicLong(0)`, since the two things that read it are `getTotalRecordsScanned()` and `runAsyncPostTransaction` both of which are `@VisibleForTesting`, and not called by any of the transactional methods, maybe that would be clearer. I was thinking that it would be nice if there was better separation between the `OnlineIndexer` that goes across transactions, and the `OnlineIndexer` that creates transactions, both because of this, and because of the `limit`.

Reply:
Created Issue: #491